django/test/testcases.py

```diff
@@ -202,13 +216,20 @@ def __call__(self, result=None):
                 self._pre_setup()
             except Exception:
                 result.addError(self, sys.exc_info())
```

Comment:
This will fail if `result` is `None`.

Reply:
Good catch! Thanks @timmartin I'll review this later (probably will add another test for this case and make sure we avoid calling `result.addError` when it's `None`)

Reply:
Hmm, it seems the default `result=None` kwarg and the call to `result.addError` are not added in this PR.

So, strictly speaking, this as an unrelated issue and, we'd probably be better discussing this in a new issue and/or patch.

However, I went ahead and mimicked [the way `unittest.suite` defines `TestSuite.debug()`][1], by making the result non-optional in our `_setup_and_call` method and by using `_DebugResult()` in our `debug()` method.

Existing method `__call__` is untouched, allowing kwarg of `result=None`, since I'm not sure whether this is used by anyone. If there are any users of this code, they will also stumble on the `AttributeError` that will be raised when calling `result.addError` above. BTW: the test suite passes by making `result` a required parameter to `__call__`, so if this is a desired change let me know and I'll update the `__call__` kwarg as well.

[1]: https://github.com/python/cpython/blob/c4d9df5fd719ad08e68e0950ce22a80f43e4f35d/Lib/unittest/suite.py#L135-L138